agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1119,27 +1119,69 @@ bool backhaul_manager::backhaul_fsm_main(bool &skip_select)
 }
 
 bool backhaul_manager::send_1905_topology_discovery_message()
+{
+    // TODO: get the list of interfaces that are up_and_running using the event-driven mechanism
+    // to be implemented in #866
+
+    /**
+     * Transmission type of Topology Discovery message is 'neighbor multicast'.
+     * That is, the CMDU must be transmitted once on each and every of its 1905.1 interfaces.
+     * Also, according to IEEE1905.1, the message should include a MAC Address TLV which contains
+     * the address of the interface on which the message is sent. Thus, a different message should
+     * be sent on each interface.
+     */
+    auto ifaces = network_utils::linux_get_iface_list_from_bridge(m_sConfig.bridge_iface);
+    for (const auto &iface_name : ifaces) {
+
+        if (!network_utils::linux_iface_is_up_and_running(iface_name)) {
+            continue;
+        }
+
+        uint32_t iface_index = network_utils::linux_get_iface_index(iface_name);
+        if (0 == iface_index) {
+            return false;
```

Comment:
Why is this a failure?
Not being able to get an interface index for its' name is indeed an issue, but what if some other interface will work?
Maybe just logging this as a WARNING and continuing to the next interface?

Reply:
Ok, I see your point and I'll adopt your suggestion.
What do you think about the rest of conditions that are also considered as a failure in this method? Shall we also log and continue?

Reply:
As long as we'll have at least one valid interface that we can send the discovery message on, I think yes.

Reply:
Done